test/test_mlir_execution.py

```diff
@@ -0,0 +1,51 @@
+from mlir.dialects.linalg.opdsl.lang import *
+from mlir.execution_engine import *
+from mlir.passmanager import *
+from mlir.dialects import linalg
+from mlir.dialects import func
+from mlir.dialects import builtin
+import sys
+import math
+import ctypes
+from micrograd.engine import Value
+from micrograd.visitor import MLIRVisitor
+# helps investigate segmentation faults
+import faulthandler
+faulthandler.enable()
+
+
+def transform(mod):
+    pm = PassManager("builtin.module", context=mod.context)
+    pm.add("func.func(convert-linalg-to-loops)")
+    pm.add("func.func(lower-affine)")
+    pm.add("func.func(convert-math-to-llvm)")
```

Comment:
I think https://mlir.llvm.org/docs/Passes/#-convert-to-llvm does all of these.

Reply:
let me try

Reply:
done